huxley/api/serializers/user.py

```diff
@@ -7,21 +7,39 @@
 
 from datetime import datetime
 
+from django.core.mail import send_mail
+from django.utils import timezone
+
 from huxley.accounts.models import User, School
 from huxley.api import validators
 from huxley.api.serializers.school import SchoolSerializer
 
 
 class UserSerializer(ModelSerializer):
-    school = SchoolSerializer(required=False, read_only=True)
+    school = SchoolSerializer(required=False)
 
     class Meta:
         model = User
         fields = ('id', 'username', 'first_name', 'last_name', 'user_type',
                   'school', 'committee',)
-        read_only_fields = ('id', 'username', 'first_name', 'last_name',
-                            'user_type', 'committee',)
+        read_only_fields = ('id', 'username', 'user_type', 'committee',)
 
+    def update(self, instance, validated_data):
+        if 'school' in validated_data:
+            school_data = validated_data.pop('school')
+            school_data['modified_at'] = timezone.now()
+            School.objects.filter(name=instance.school).update(**school_data)
+            send_mail('{0} has updated its registration information'.format(instance.school),
```

Comment:
Update needs to be called explicitly on the school here because django's default update method doesn't handle nested serializers